src/rgw/rgw_log.cc

```diff
@@ -200,25 +200,21 @@ static void log_usage(req_state *s, const string& op_name)
   if (!usage_logger)
     return;
 
-  std::string user;
+  std::string user = to_string(s->owner.id);
   std::string payer;
-  string bucket_name;
-
-  bucket_name = s->bucket_name;
+  string bucket_name = s->bucket_name;
 
   if (!bucket_name.empty()) {
-    bucket_name = s->bucket_name;
     user = to_string(s->bucket_owner.id);
+
     if (!rgw::sal::Bucket::empty(s->bucket.get()) &&
-	s->bucket->get_info().requester_pays) {
+        s->bucket->get_info().requester_pays &&
+        s->err.http_ret != 403) {
       payer = s->user->get_id().to_str();
     }
-  } else {
-    user = to_string(s->owner.id);
   }
 
-  bool error = s->err.is_err();
-  if (error && s->err.http_ret == 404) {
+  if (rgw::sal::Bucket::empty(s->bucket.get())) {
```

Comment:
why did this need to change?

Reply:
It's a bit confusing - I guess a 404 HTTP status code could also mean an object was not found or any other related to the action and doesn't necessarily mean the bucket is not found and if we have the bucket in `s->bucket` it means the bucket is valid. Am I mistaken?

Reply:
i don't know why this needs to do anything other than `bucket_name = s->bucket_name;` above, really

but this seems unrelated to the request payer change. it should be a separate commit, at least

Reply:
I guess it makes sense to have it not flood the usagelog with invalid buckets (as each bucket would result in a unique `rgw_usage_log_entry`).
I'll remove it from this PR and propose a new PR so we can discuss it there. It's irrelevant from this PR and ticket as you said.

Reply:
ok. i didn't even realize this was for the usage log, i thought it was for the ops log